src/codewhisperer/service/securityIssueHoverProvider.ts

```diff
@@ -41,6 +41,44 @@ export class SecurityIssueHoverProvider implements vscode.HoverProvider {
         return new vscode.Hover(contents)
     }
 
+    /**
+     * Updates the position of each hover when the text document is changed. The underlying security issues
+     * will be updated with new line numbers to reflect the new positions of the hover.
+     *
+     * @param event Event that triggered the text document change
+     */
+    public updateRanges(event: vscode.TextDocumentChangeEvent) {
+        const changedRange = event.contentChanges[0].range
+        const changedText = event.contentChanges[0].text
+        const lineOffset = this._getLineOffset(changedRange, changedText)
+
+        this._issues = this._issues.map(issues => this._applyRangeOffset(event.document.fileName, issues, lineOffset))
+    }
+
+    private _getLineOffset(range: vscode.Range, text: string) {
+        const originLines = range.end.line - range.start.line + 1
+        const changedLines = text.split('\n').length
```

Comment:
Please verify if this split is not extra splitting on any `\n` which user has written as part of any string variables.

Reply:
Verified it is working as expected